frontend/src/routes/password.change.tsx

```diff
@@ -0,0 +1,293 @@
+        );
+      case SetPasswordStatus.NoCurrentPassword:
+        return t(
+          "frontend.password_change.failure.description_NO_CURRENT_PASSWORD",
+        );
+      case SetPasswordStatus.NotAllowed:
+        return t("frontend.password_change.failure.description_NOT_ALLOWED");
+      case SetPasswordStatus.NotFound:
+        return t("frontend.password_change.failure.description_NOT_FOUND");
+      case SetPasswordStatus.PasswordChangesDisabled:
+        return t(
+          "frontend.password_change.failure.description_PASSWORD_CHANGES_DISABLED",
+        );
+      case SetPasswordStatus.WrongPassword:
+        return t("frontend.password_change.failure.description_WRONG_PASSWORD");
+      default:
+        return undefined;
+    }
+  })();
+
+  return (
+    <Layout>
+      <BlockList>
+        <PageHeading
+          Icon={IconLockSolid}
+          title={t("frontend.password_change.title")}
+          subtitle={t("frontend.password_change.subtitle")}
+        />
+
+        <Form.Root onSubmit={onSubmit} method="POST">
+          {/*
+            In normal operation, the submit event should be `preventDefault()`ed.
+            method = POST just prevents sending passwords in the query string,
+            which could be logged, if for some reason the event handler fails.
+          */}
+
+          {!success && (
+            <>
+              <Form.Field
+                name="current_password"
+                serverInvalid={
+                  result.data &&
+                  result.data.setPassword.status ==
+                    SetPasswordStatus.WrongPassword
+                }
+              >
+                <Form.Label>
+                  {t("frontend.password_change.current_password_label")}
+                </Form.Label>
+
+                <Form.PasswordControl
+                  required
+                  autoComplete="current-password"
+                  ref={currentPasswordRef}
+                />
+
+                <Form.ErrorMessage match="valueMissing">
+                  {t("frontend.errors.field_required")}
+                </Form.ErrorMessage>
+
+                {result.data &&
+                  result.data.setPassword.status ==
+                    SetPasswordStatus.WrongPassword && (
+                    <Form.ErrorMessage>
+                      {t(
+                        "frontend.password_change.failure.description_WRONG_PASSWORD",
+                      )}
+                    </Form.ErrorMessage>
+                  )}
+              </Form.Field>
+
+              <Separator />
+
+              <Form.Field name="new_password">
+                <Form.Label>
+                  {t("frontend.password_change.new_password_label")}
+                </Form.Label>
+
+                <Form.PasswordControl
+                  required
+                  autoComplete="new-password"
+                  ref={newPasswordRef}
+                  onBlur={() => newPasswordAgainRef.current!.reportValidity()}
```

Comment:
I don't really like that it triggers on the first run. Maybe check whether the value is non empty? 
```suggestion
                  onBlur={() => newPasswordAgainRef.current!.value && newPasswordAgainRef.current!.reportValidity()}
```